### DISCUSSION TITLE
`Paid` Label

### DISCUSSION THREAD

**Discussion by 0x4007** [ubiquity/ubiquibot#255] [Polls]
<img width="543" alt="image" src="https://user-images.githubusercontent.com/4975670/231121739-eb92ba85-8018-4315-81b7-9cd2f44d917d.png">

@Steveantor created a new label, `Paid` which is used to check if a permit was generated for the bounty. This is a significant decision that affects our architecture due to the way the label is used. This slipped in past @rndquu and my [review](https://github.com/ubiquity/bounty-bot/pull/226/files#r1162506793) without notice. I was quick to make a [hot fix issue](https://github.com/ubiquity/bounty-bot/issues/253) to remove it but @Steveantor has some points about why we should keep it. 

1.  @Steveantor primarily suggests that we can filter by the label to reduce API calls. 
2. An example of how we would normally check for permits is by looping through every comment of every issue. I think @rndquu has the most insight on this given their implementation on the [audit tool](https://github.com/ubiquity/pay.ubq.fi/issues/22) the other day. We have a [new capability](https://github.com/ubiquity/bounty-bot/issues/134) for the bot to be aware of rate limits and act accordingly. Further, the more contributors that are on a GitHub repository, the higher the rate limits are. lastly, we're barely in beta so I think that we shouldn't concern ourselves with large initiatives around rate limit optimization just yet.
3. Lastly my point for why it should be removed is that it is misleading. More information [here](https://github.com/ubiquity/bounty-bot/issues/158#issuecomment-1502911861).

I would like to make the optimal decision (remove, add, modify) by including the opinions of the contributors.

**Comment by 0x4007** [comment]
I think that `1. ` has merit but I also am not very concerned with rate limit optimization until we're closer or already at "v1" of the bot. `2. ` Manually checking every comment on every issue, guarantees that the data integrity will be highest and the system will be the most robust (unless of course, we are rate limited!) It also is the architecture that aligns the most with if we plan to use GitHub (comments) as our bot's database (the bot will need to already look through every comment for other features like wallet registrations etc.)
`3. ` We could rename it to `Permitted` if we plan to keep it.

**Comment by rndquu** [comment]
These 2 checks: [first](https://github.com/ubiquity/bounty-bot/pull/226/files#diff-d654457614336d178b63b012cb2a680a783519bd083636d4b0bdf7bafef93809R24) and [second](https://github.com/ubiquity/bounty-bot/pull/226/files#diff-d654457614336d178b63b012cb2a680a783519bd083636d4b0bdf7bafef93809R36) are basically equal so ideally we should keep only 1 of those.

We have 2 options:
1. Remove the `Paid` label, remove this [check](https://github.com/ubiquity/bounty-bot/pull/226/files#diff-d654457614336d178b63b012cb2a680a783519bd083636d4b0bdf7bafef93809R24) and check if permit has been generated by parsing issue comments. This way we utilize a greater number of API requests. 
2. Remove the following checks: [1st](https://github.com/ubiquity/bounty-bot/pull/226/files#diff-d654457614336d178b63b012cb2a680a783519bd083636d4b0bdf7bafef93809R30) and [2nd](https://github.com/ubiquity/bounty-bot/pull/226/files#diff-d654457614336d178b63b012cb2a680a783519bd083636d4b0bdf7bafef93809R36), keep using the label (which reduces number of API requests) and rename the `Paid` label.

For me adding the `Paid` label is not a significant change but I agree that the name is misleading. When we have gnosis safe intergrated we could have a "paid" label for both cases: permit2 + gnosis safe transaction. I would go with option 2: remove unnecessary checks and rename the `Paid` label to smth like `Payment request created`.

**Reply by 0x4007** [reply]
`Payment Authorized`
